plugins/woocommerce/includes/admin/helper/class-wc-woo-update-manager-plugin.php

```diff
@@ -108,6 +108,25 @@ public static function show_woo_update_manager_install_notice(): void {
 		include __DIR__ . '/views/html-notice-woo-updater-not-activated.php';
 	}
 
+	/**
+	 * Adjust update counter if Woo.com Update Manager is not installed.
+	 *
+	 * @param int $count number of updates available excluding Woo.com Update Manager.
+	 * @return int
+	 */
+	public static function increment_update_count_for_woo_update_manager( int $count ): int {
+
+		if ( ! WC_Helper::is_site_connected() ) {
+			return 1;
+		}
+
+		if ( ! self::is_plugin_installed() || ! self::is_plugin_active() ) {
+			++$count;
+		}
+
+		return $count;
+	}
```

Comment:
This method name doesn't accurately reflect what the code is actually doing. It says "increment", but it actually overwrites some case. 

Also, if there is a condition when the count is overwritten, it's probably best to call it in the first place, preventing execution of the earlier calculations. Given that this is run on every request admin request, it's worth considering performance implications.

Reply:
Fixed here: https://github.com/woocommerce/woocommerce/pull/46088/commits/f09891b9b7f1beca9088881dae7d969b2bf50bc9